src/state/routing/utils.ts

```diff
@@ -269,17 +255,16 @@ export async function transformRoutesToTrade(
     gasUseEstimateUSD,
     approveInfo,
     blockNumber,
-    isUniswapXBetter,
     requestId: data.quote.requestId,
     quoteMethod,
     inputTax,
     outputTax,
     swapFee,
   })
 
-  // During the opt-in period, only return UniswapX quotes if the user has turned on the setting,
-  // even if it is the better quote.
-  if (isUniswapXBetter && (routerPreference === RouterPreference.X || isUniswapXDefaultEnabled)) {
+  // If the top-level URA quote type is DUTCH_LIMIT, then UniswapX is better for the user
+  const isUniswapXBetter = data.routing === URAQuoteType.DUTCH_LIMIT
+  if (isUniswapXBetter) {
```

Comment:
this could be in a follow up but we could re-organize this code here since we can always just show the user the `trade` type that is the top-level quote 

and we dont have to prepare the `UniswapXTrade` if the top level quote is just classic

Reply:
i'll follow up with this change since the PR is already big